Automattic/jetpack
# Hide the Customizer menu for block themes

Closes https://github.com/Automattic/dotcom-forge/issues/5778
Related to: 
- https://github.com/Automattic/wp-calypso/pull/87973
- pfsHM7-fU-p2#comment-393
- https://github.com/Automattic/jetpack/pull/27238
- https://github.com/Automattic/wp-calypso/issues/60732
- https://github.com/Automattic/jetpack/pull/23670

## Proposed changes:


This PR aims to hide the Customizer menu when a block theme is active.

<img width="456" alt="Screenshot 2024-02-29 at 10 41 35" src="https://github.com/Automattic/jetpack/assets/5287479/fb29853c-a59d-4136-8f19-5a01c7ff3eee">

- The approach is to remove all actions added on WordPress.com, as discussed in p58i-f7m-p2#comment-59243. The rationale for this approach is detailed in the same post (p58i-f7m-p2), noting that the majority of these actions are features designed for classic themes and are not necessary when a block theme is in use. 
- There is one thing that should be utilized even in block themes, so I have submitted a PR: https://github.com/Automattic/wp-calypso/pull/87973. We need to update the doc afterward: https://wordpress.com/support/footer-credits/.
- Note that the Customizer menu is shown if a **3rd party plugin** depends on Customizer (this PR only removes actions from the dotcom side). This is the [core behavior](https://github.com/wordpress/wordpress-develop/blob/64b699e8e31fbff60c1a4cf51a70cbdcbb794d7b/src/wp-admin/menu.php#L219) and a reasonable decision as seen in https://github.com/Automattic/wp-calypso/issues/60732, for example.

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Create a new site
* Patch this change: https://github.com/Automattic/jetpack/pull/36017#issuecomment-1968478186
* Go to any Calypso page
* Ensure the Customizer menu is hidden
* Go to any wp-admin page
* Ensure the Customizer menu is hidden
* Switch to a Classic theme (e.g., Twenty Eleven)
* Ensure the Customizer menu is shown
